lib/request.ts

```diff
@@ -51,16 +52,22 @@ export default class CIORequest {
     const headers = {
       Authorization: this.auth,
       'Content-Type': 'application/json',
-      'Content-Length': body ? Buffer.byteLength(body,'utf8') : 0,
+      'Content-Length': body ? Buffer.byteLength(body, 'utf8') : 0,
     };
 
     return { method, uri, headers, body };
   }
 
   handler({ uri, body, method, headers }: RequestHandlerOptions): Promise<Record<string, any>> {
     return new Promise((resolve, reject) => {
-      let options = Object.assign({}, this.defaults, { method, headers });
-      let req = request(uri, options, (res) => {
+      let url = new URL(uri);
```

Comment:
This doesn't appear to work with [the npm module `"url"`](https://www.npmjs.com/package/url), which is what we're using in an RN environment (despite that describing itself as `meant to have feature parity with node.js core url module.`).

I'm getting `undefined is not a constructor` because `URL` isn't a property. Logging the entire import shows:

```
{
  "Url": [Function Url],
  "format": [Function urlFormat],
  "parse": [Function urlParse],
  "resolve": [Function urlResolve],
  "resolveObject": [Function urlResolveObject]
}
```

Reply:
Yikes, and after swapping `new URL(…)` for `url.parse(…)`, I'm getting `Unable to parse TLS packet header` from the request!! I'll see if I can get something working, the RN environment seems especially finicky.

Reply:
@vcarl yeah, our customerio-node package is mainly meant to target Node.js backends. It may be easier for you to use our API directly: https://customer.io/docs/api/
It would be a little bit more setup and you'd have to be sending the requests yourself, but then you wouldn't need to worry about Node vs React Native incompatibilities.
In future we would like to offer an SDK for React Native and other mobile platforms, though we don't have a specific timeline on it yet.